File: editor-server/src/resolvers/subscriptions/positionMap.ts
import { ObjectType, Field, ID } from "type-graphql";
import { PosData, PosDataScalar } from "../types/posData";

@ObjectType()
export class PositionMapPayload {
  @Field((type) => PosDataScalar)
  frame: any;

  @Field()
  editBy: number;
}

File: editor-server/src/graphql/subscriptions/position_map.rs
//! PositionMap subscription methods.

use crate::graphql::subscriptor::Subscriptor;
use crate::graphql::types::pos_data::PosDataScalar;
use async_graphql::{SimpleObject, Subscription};
use futures_core::stream::Stream;

#[derive(SimpleObject, Clone, Default)]
pub struct PositionMapPayload {
    pub edit_by: i32,
    pub frame: PosDataScalar,
}

#[derive(Default)]
pub struct PositionMapSubscription;

#[Subscription]
impl PositionMapSubscription {
    async fn position_map_subscription(&self) -> impl Stream<Item = PositionMapPayload> {
        Subscriptor::<PositionMapPayload>::subscribe()
    }
}
